### PR TITLE
Migrate `kontrol` module from `evm-semantics`

### PR DESCRIPTION
**Pull request by tothtamas28**
* Copies `kontrolx` files from https://github.com/runtimeverification/evm-semantics/commit/a9231d467aacbdf987a0d2304bf7361c8d135e05
  * Renames module: `kontrolx => kontrol`
  * Renames commands: `kontrolx foundry-kompile => kontrol build`, `kontrolx foundry-X => kontrol X`
* Adds `Dockerfile`
* Sets up `flake.nix` (thanks @goodlyrottenapple)
* Adds CI jobs for testing: `test-pr.yml`
* Adds CI job for updating dependencies: `update-version.yml`

### CODE REVIEW

File: src/tests/integration/test-data/foundry-prove-all [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+AccountParamsTest.test_getNonce_unknownSymbolic(address)
+AccountParamsTest.test_Nonce_ExistentAddress()
+AccountParamsTest.test_Nonce_NonExistentAddress()
+AccountParamsTest.testNonceSymbolic(uint64)
+AddrTest.test_addr_false()
+AddrTest.test_addr_symbolic(uint256)
+AddrTest.test_addr_true()
+AddrTest.test_builtInAddresses()
+AddrTest.testFail_addr_false()
+AddrTest.testFail_addr_true()
```
Comment (ehildenb) [inline src/tests/integration/test-data/foundry-prove-all:18]: Are we able to enumerate this list instead by inspecting the foundry kompiled output? That would be good for making sure that we don't accidentally not enable a newly added test.

Either that, or add a simple test-step which just compares this list against foundry compiled output to make sure that it's complete.

**Comment by ehildenb** [comment]
We need to make sure to include a Nix flake so that people can install `kontrol` via Kup.

File: src/kontrol/__main__.py [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+    print(k_text)
+
+
+def exec_foundry_kompile(
+    foundry_root: Path,
+    includes: Iterable[str] = (),
+    regen: bool = False,
+    rekompile: bool = False,
+    requires: Iterable[str] = (),
+    imports: Iterable[str] = (),
```
Comment (ehildenb) [inline src/kontrol/__main__.py:113]: We should rename the commands to be:

- `foundry-kompile => build`
- All other command sremove the `foundry` prefix.

Comment (tothtamas28) [reply src/kontrol/__main__.py]: Done.

**Review by ehildenb — APPROVED** [review]
Two things that should become new issues:

- We should be publishing the docker images to `dockerhub`.
- We should call `forge build` as part of `kontrol build`, so that users only have to remember `kontrol ...` command. They of course can still call `forge build` themselves (`forge` is pretty good at calculating if it needs to actually rebuild anything).
- Add `kontrol` to `kup` listing (PR into `kup`).
- Add `kontrol` to our automation (make sure it's included in the automerged repos).

File: .github/workflows/test-pr.yml [hunk 3]
```diff
@@ -58,19 +58,116 @@ jobs:
...
+          docker stop --time=0 kontrol-ci-profile-${GITHUB_SHA}
+
   integration-tests:
     needs: code-quality-checks
     name: 'Integration Tests'
     runs-on: [self-hosted, linux, normal]
+    strategy:
+      matrix:
+        backend: ['legacy', 'booster']
+    timeout-minutes: 150
```
Comment (ehildenb) [inline .github/workflows/test-pr.yml:100]:
```suggestion
    timeout-minutes: 240
```

**Comment by tothtamas28** [comment]
> things that should become new issues

Done:
* https://github.com/runtimeverification/kontrol/issues/31
* https://github.com/runtimeverification/kontrol/issues/32
* https://github.com/runtimeverification/kontrol/issues/33
* https://github.com/runtimeverification/kontrol/issues/34

**Pull request merged**